system variable: refine description of txn-total-size-limit and tidb_mem_quota_query (#11530)
pingcap/docs: tidb-configuration-file.md

- The size limit of a single transaction in TiDB.
- Default value: `104857600` (in bytes)
- In a single transaction, the total size of key-value records cannot exceed this value. The maximum value of this parameter is `1099511627776` (1 TB). Note that if you have used the binlog to serve the downstream consumer Kafka (such as the `arbiter` cluster), the value of this parameter must be no more than `1073741824` (1 GB). This is because 1 GB is the upper limit of a single message size that Kafka can process. Otherwise, an error is returned if this limit is exceeded.

> - In TiDB v6.5.0 and later versions, this configuration is no longer recommended. The memory size of a transaction will be accumulated into the memory usage of the session, and the [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) variable will take effect when the session memory threshold is exceeded. To be compatible with previous versions, the behavior of this configuration is as follows when you upgrade from an earlier version to TiDB v6.5.0 or a later version:

Reviewer: ```suggestion
- In TiDB v6.5.0 and later versions, this configuration is no longer recommended. The memory size of a transaction will be accumulated into the memory usage of the session, and the [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) variable will take effect when the session memory threshold is exceeded. To be compatible with previous versions, this configuration works as follows when you upgrade from an earlier version to TiDB v6.5.0 or later:
```

>     - If this configuration is not set or is set to the default value (104857600), after the upgrade, the memory size of the transaction will be accumulated into the memory usage of the session, and the `tidb_mem_quota_query` variable will take effect.

Reviewer: ```suggestion
    - If this configuration is not set or is set to the default value (`104857600`), after an upgrade, the memory size of a transaction will be accumulated into the memory usage of the session, and the `tidb_mem_quota_query` variable will take effect.
```

>     - If the value of this configuration is not the default value (104857600), the behavior of the configuration to limit the size of a single transaction remains unchanged before and after the upgrade. The configuration still takes effect, and the memory size of the transaction is not controlled by the `tidb_mem_quota_query` variable.

Reviewer: ```suggestion
    - If this configuration is not defaulted (`104857600`), it still takes effect and its behavior on controling the size of a single transaction remains unchanged before and after the upgrade. This means that the memory size of the transaction is not controlled by the `tidb_mem_quota_query` variable.
```
